tests/cxx/template_default_args_roundtrip.cc

```diff
@@ -0,0 +1,32 @@
+//===--- template_default_args_roundtrip.cc - test input file for iwyu ----===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+
+#include "template_default_args_roundtrip-template.h"
+
+//Provides DefaultArgument<int> indirectly
+#include "template_default_args_roundtrip-indirect.h"
+//Provides DefaultArgument<int> directly
+#include "template_default_args_roundtrip-direct.h"
+
+//Template making a full use of DefaultArgument<int>
+Template<int> templ;
```

Comment:
Is there an IWYU diagnostic on this line you could assert?

Reply:
There is no IWYU diagnostic on purpose because the template is a full use. If it weren't, IWYU would issue a diagnostic and the test would fail.

Reply:
Makes sense.